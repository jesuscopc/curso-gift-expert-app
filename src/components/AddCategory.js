import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorie }) => {

  const [inputValue, setInputValue] = useState('');
  
  const handleChangeInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputValue.trim().length > 2) {
      setCategorie( [inputValue] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <h1>{ inputValue }</h1>
      <input
        type="text"
        value={ inputValue }
        onChange={ handleChangeInputValue }
      />
    </form>
  )
};

AddCategory.propTypes = {
  setCategorie: PropTypes.func.isRequired,
};