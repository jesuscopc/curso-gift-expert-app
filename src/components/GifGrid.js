import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categorie }) => {

  const { data: imgs, loading } = useFetchGifs( categorie );

  return (
    <>
      <h3 className="card animate__animated animate__fadeIn">{ categorie }</h3>
      { loading && <p className="card animate__animated animate__flash">loading...</p> }
      <div className="card-grid">
          {imgs.map( img =>
            (
              <GifGridItem
                key={ img.id }
                { ...img } 
              />
            )
          )}
      </div>
    </>
  )
}
