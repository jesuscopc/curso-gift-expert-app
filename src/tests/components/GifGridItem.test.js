import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Sui Test GifGridItem', () => {

  const title = 'Titulo';
  const url = 'http://localhost:algo.jpg';
  const wrapper = shallow(
    <GifGridItem
      title={title}
      url={url}
    />
  );

  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render title correctly', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  })

  test('should be render correctly props img', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should be have class animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.props('className').className;
    expect(className.includes('animate__fadeIn')).toBe(true);
  })
  
  
  
  
});
