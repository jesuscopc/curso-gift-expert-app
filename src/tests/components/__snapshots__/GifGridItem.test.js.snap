// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui Test GifGridItem should be render correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="http://localhost:algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
